src/commands/githubActions/bumpUpstream/setup.ts

```diff
@@ -0,0 +1,99 @@
+import { Manifest, Compose } from "@dappnode/types";
+import { readManifest, readCompose } from "../../../files/index.js";
+import { arrIsUnique } from "../../../utils/array.js";
+import { parseCsv } from "../../../utils/csv.js";
+import { readBuildSdkEnvFileNotThrow } from "../../../utils/readBuildSdkEnv.js";
+import { GitSettings, InitialSetupData, UpstreamSettings } from "./types.js";
+import { getFirstAvailableEthProvider } from "../../../utils/tryEthProviders.js";
+
+export async function readInitialSetup({ dir, userEthProvider, useFallback }: { dir: string, userEthProvider: string, useFallback: boolean }): Promise<InitialSetupData> {
+    const envFileArgs = readBuildSdkEnvFileNotThrow(dir);
+
+    const { manifest, format } = readManifest({ dir });
+    const compose = readCompose({ dir });
+
+    const upstreamRepos = envFileArgs
+        ? [envFileArgs._BUILD_UPSTREAM_REPO]
+        : parseCsv(manifest.upstreamRepo);
```

Comment:
could you add a todo with a link reference to https://github.com/dappnode/DAppNodeSDK/issues/408 saying that the upstreamRepo format will change